### ISSUE TITLE
aiohttp based asyncio client

### ISSUE THREAD

**Issue by graingert** [google/google-api-python-client#360]
I'd like to use this in a Sanic service in our platfrom, however I'm having to use run_in_executor() to use this library without stalling the eventloop.

It would make life faster and easier if this library would also support asyncio.

**Comment by jonparrott** [comment]
This is pretty unlikely to happen in this particular library- it's too encumbered by httplib2 and the burden of supporting 2.x it going to make it difficult.

It would be super interesting to see, though. :)

**Comment by graingert** [comment]
@jonparrott it would be possible to support python 2.x and python 3.x at the same time by using a custom deferred, future or IOAction implementation.

**Comment by jonparrott** [comment]
@graingert yeah but at that point we'd have a completely different library.

**Comment by graingert** [comment]
I don't think it would be that different, a lot of the api classes just transform data without performing further IO.

**Comment by jonparrott** [comment]
It's still a lot more than I think this library could take, and a major version bump would cause lots of issues. We went through this exercise with oauth2client and ended up just needing to write a new library altogether. This library is really old, has a lot of old idioms, and is very tied to the way its http library does things. It's just not something we can commit any time to here.

We're largely investing in the new google-cloud-python library. I'd suggest requesting this feature there. Alternatively, this is open source so you're free to make an async fork - but you'd be better off starting from scratch. :)

**Comment by graingert** [comment]
@jonparrott https://github.com/GoogleCloudPlatform/google-cloud-python/issues/3103

**Comment by froosty** [comment]
I came across this question when I was looking an asynchronous library for Google Analytics. My message is off topic a little bit, but maybe it helps to someone. We created an asynchronous client for Google Analytics (measurement protocol) and it can be found here: https://github.com/wikibusiness/aioga. It very simple for now, but in nearest future, we plan to expand it.

**Comment by thehesiod** [comment]
@jonparrott @froosty let me know if you ever find an async client.  We have a lambda that users google APIs so doing as many calls as possible in parallel would be nice.  Looks like there are some oauth clients so that can help.
